### PR TITLE
docs(udon-sharp): use direct SDK 3.10.4 event receivers

### LINKED ISSUE #342: Document that UdonSharpBehaviour implements IUdonEventReceiver in SDK 3.10.4
### Category

New SDK feature coverage

### Target Skill

unity-vrc-udon-sharp

### Failure basis

Technical — silent failure, compile error, data desync, API ignore, memory leak, etc.

## 概要

VRChat Worlds SDK 3.10.4から、`UdonSharpBehaviour` が `IUdonEventReceiver` を直接実装するようになりました。

公式リリースノートにも、次の変更が記載されています。

> `UdonSharpBehaviour` now implements `IUdonEventReceiver`, saving you some type casting.

## SDK 3.10.4以降

```csharp
VRCStringDownloader.LoadUrl(url, this);
```

```csharp
imageDownloader.DownloadImage(
    url,
    material,
    this,
    textureInfo
);
```

明示的なキャストが不要なため、ほかに `IUdonEventReceiver` を参照していなければ、次のusingも不要になります。

```csharp
using VRC.Udon.Common.Interfaces;
```

## SDK 3.10.3以前

SDK 3.10.3以前の `UdonSharpBehaviour` は `IUdonEventReceiver` を直接実装していないため、従来の明示的なキャストが必要です。

```csharp
VRCStringDownloader.LoadUrl(
    url,
    (IUdonEventReceiver)this
);
```

## 現在のskillsの記述

現在のskillsでは、SDKバージョンに関係なく、次の記述が一貫して使用されています。

```csharp
VRCStringDownloader.LoadUrl(
    url,
    (IUdonEventReceiver)this
);
```

`references/web-loading.md` のトラブルシューティングには、Image Loadingについて次の記述もあります。

```text
Explicitly pass `(IUdonEventReceiver)this`
```

ここでは、以下の2つが区別されていません。

1. 完了イベントの受信先としてreceiver引数を明示的に渡すこと
2. receiverを `IUdonEventReceiver` へ明示的にキャストすること

SDK 3.10.4以降でもreceiver引数自体は必要ですが、`IUdonEventReceiver` への明示的なキャストは不要です。

## 追加してほしい内容

以下のバージョン差分をskillsへ追加することを提案します。

- SDK 3.10.4から `UdonSharpBehaviour` が `IUdonEventReceiver` を直接実装する
- SDK 3.10.4以降では、`IUdonEventReceiver` を要求するAPIへ `this` を直接渡せる
- SDK 3.10.3以前では、従来の `(IUdonEventReceiver)this` が必要
- String/Image Loadingのreceiver引数自体は、SDK 3.10.4以降も引き続き必要
- 複数のSDKバージョンを対象とするコード例では、従来の明示的キャストを残すことで互換性を維持できる
- SDK 3.10.4専用のコードでは、不要なキャストとusingを省略できる

### Official Documentation URL

https://creators.vrchat.com/releases/release-3-10-4/

### Code Example

```csharp
// SDK 3.10.4以降
VRCStringDownloader.LoadUrl(url, this);

// SDK 3.10.3以前との互換性が必要な場合
VRCStringDownloader.LoadUrl(url, (IUdonEventReceiver)this);
```

### PR DESCRIPTION
**Pull request by niaka3dayo**
## Summary

- Update active UdonSharp examples for SDK 3.10.4 so `UdonSharpBehaviour` is passed directly as `this` while the required receiver argument remains present.
- Keep the pre-3.10.4 cast in the historical migration reference only, and add a documentation smoke test covering the active examples, receiver arguments, namespace imports, and migration boundary.
- Correct the active image-loading namespace to `VRC.SDK3.Image`, based on the SDK 3.10.4 source and wrapper API.

## Validation

- All documentation smoke tests
- Bash syntax checks for repository shell scripts
- `validate-udonsharp` shell and BusyBox smoke tests
- Changed Markdown C# blocks through `validate-udonsharp.sh`
- `git diff --check`
- Symlink check
- `npm pack --dry-run`
- Version consistency check

## Unity verification

The first two local attempts were not successful and remain part of the diagnostic history: the WSL/UNC project was rejected because Unity does not support the case-sensitive filesystem (exit 21), and the first Windows-local probe used the obsolete `VRC.SDK3.ImageLoading` / `IVRCImageDownload` names (exit 1). No third run was made from this worktree after that stop condition.

A separate Windows-local verification was completed by the parent workflow after correcting the fixture and rebuilding the copied project. Unity 2022.3.22f1 with SDK 3.10.4 completed the Tundra build and UdonSharp compilation of three fixtures, reported `ISSUE342_PASS`, and confirmed direct `this` for String, Image, and `NetworkCalling` receiver calls. The result also confirmed `UdonSharpBehaviour` implements `IUdonEventReceiver` and the image API is `VRC.SDK3.Image.VRCImageDownloader`; source and execution copies matched by hash.

The same invocation initially reported a stale Bee input referencing unrelated removed Issue #338 utility scripts (`CS2001`). An additional Tundra rebuild followed, and only the subsequent successful build/compile/result is counted as verification. Compilation verification covers the three targeted fixtures; it does not claim runtime execution or unrelated scripts.

Closes #342

### CODE REVIEW

**Comment by niaka3dayo** [comment]
@coderabbitai review

File: skills/unity-vrc-udon-sharp/references/web-loading.md [hunk 1]
```diff
@@ -6,12 +6,26 @@ Older version numbers in this reference record feature introductions or migratio
 
 Since `System.Net` is unavailable in UdonSharp, VRChat-specific APIs must be used to retrieve data from the web.
 
+### SDK 3.10.4 event receiver arguments
```
[the remark this answers is not in the document]
Comment (niaka3dayo) [reply skills/unity-vrc-udon-sharp/references/web-loading.md]: ありがとうございます。見出し階層が崩れていました。`##` に直し、markdownlintでも確認しました。対応: aaebe6c

File: skills/unity-vrc-udon-sharp/SKILL.md [hunk 2]
```diff
@@ -55,6 +55,18 @@ compiler constraints, use `unity-vrc-world-sdk-3` and read
...
+
+SDK 3.10.4 is the active and verified target for this Skill.
+SDK 3.10.4: `UdonSharpBehaviour` implements `IUdonEventReceiver` directly.
+An API that requires a receiver can therefore receive `this` directly:
+
+```csharp
+VRCStringDownloader.LoadUrl(dataUrl, this);
+```
+
+The receiver argument is still required; only the explicit `(IUdonEventReceiver)` cast is unnecessary on the active SDK. Keep any pre-3.10.4 cast in the historical migration reference only. This follows the [official SDK 3.10.4 release notes](https://creators.vrchat.com/releases/release-3-10-4/) and the SDK source declaration.
```
[the remark this answers is not in the document]
Comment (niaka3dayo) [reply skills/unity-vrc-udon-sharp/SKILL.md:68]: ありがとうございます。3.10.4の変更点として漏れていたので、SDK表にも`UdonSharpBehaviour`のreceiver対応を追記しました。対応: aaebe6c

File: tests/docs/udon-event-receiver-reference.test.sh [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+if missing_receivers:
+    raise SystemExit(
+        "receiver argument missing or not direct this in active examples:\n"
+        + "\n".join(missing_receivers)
+    )
+
+print("PASS: active receiver examples use direct this with an explicit receiver argument")
+print("PASS: historical casts remain isolated to sdk-migration.md")
+print("PASS: unused IUdonEventReceiver namespace imports are absent from active blocks")
+PY
```
[the remark this answers is not in the document]
Comment (niaka3dayo) [reply tests/docs/udon-event-receiver-reference.test.sh:233]: ここは追加しません。`validate-udonsharp.sh` はUdonSharpの構文制約を調べる静的フックで、今回追加したreceiver契約やSDK APIの型整合性は検証できません。対象3例はSDK 3.10.4上でUdonSharpコンパイル済みで、Markdown内の変更例もローカルでフックを通しています。docs CIに同じ内容のstandalone fixtureを持つと実例との二重管理になるため、現在の契約テストとUnityコンパイル証拠を正本にします。対応: aaebe6c
